src/drivers/mct/main/component_mod.F90

```diff
@@ -377,9 +386,20 @@ subroutine component_init_cx(comp, infodata)
                 if (iamroot_CPLID) then
                    write(logunit,F0I) 'creating dom_cx'
                    call shr_sys_flush(logunit)
-                end if
+                end igit cof
                 call seq_mctext_gGridInit(comp(1))
-                call seq_map_map_exchange(comp(1), flow='c2x', dom_flag=.true., msgtag=comp(1)%cplcompid*10000+1*10+1)
+
+                if (size(comp) > 1) then
+                    mpi_tag = comp(eci)%cplcompid*100+eci*10+1
```

Comment:
Why two cases here?  Why not just use line 393 regardless of size(comp)?

Reply:
To summarize the thread, that was the suggestion by @amametjanov to multi-instance runs with >20 instance to run without exceeding the MPI tag limit, but still prevent tag collision on single instance high res runs. Multi-driver is a possible alternative, but there were challenges around setting that up for people in this thread and others I've talked to about this issue. It seems if multi-instance is going to be an option it should be able to handle at least a normal ensemble size. 

I'd be welcome to a suggestion on how to assign these tag values entirely differently, but I don't know them, or with what they are colliding, well enough. I suggested re-ordering the way the tags were written, but that didn't get any traction from anyone in this thread. So this at least fixes the immediate issue.